docs/docs/design/parachain-protocol/bitfield-distribution.md

```diff
@@ -0,0 +1,130 @@
+`DistributeBitfield` to ensure only valid bitfields are sent to peers.
+
+Additionally, the subsystem must handle the following general network bridge events and overseer signals:
+
+1. `networkbridgevent.PeerConnected`
+2. `networkbridgevent.PeerDisconnected`
+3. `networkbridgevent.NewGossipTopology`
+4. `networkbridgevent.PeerViewChange`
+5. `networkbridgevent.OurViewChange`
+6. `networkbridgevent.UpdatedAuthorityIDs`
+7. `parachaintypes.ActiveLeavesUpdateSignal`
+
+The overseer must be modified to forward these messages to the subsystem.
+
+### Messages Sent
+
+When handling `DistributeBitfield` messages, the subsystem sends a `ProvisionableDataBitfield` message to the overseer.
+This type should be added in `dot/parachain/provisioner/messages/messages.go` in the following way:
+
+```go
+package provisionermessages
+
+import (
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+var (
+	// ...
+	_ Data = (*ProvisionableDataBitfield)(nil)
+)
+
+// ...
+
+type ProvisionableDataBitfield struct {
+	RelayParent common.Hash
+	Bitfield    parachaintypes.UncheckedSignedAvailabilityBitfield
+}
+
+func (ProvisionableDataBitfield) IsData() {}
+```
+
+If the type `CheckedSignedAvailabilityBitfield` mentioned above is created, it should also be used in
+`ProvisionableDataBitfield`.
+
+The subsystem also sends `networkbridgemessages.ReportPeer` during handling of various messages.
+
+## Subsystem State
+
+The subsystem should store the view of each peer the subsystem is informed about via the relevant network bridge events.
+The Parity node also stores [the protocol version of the peer](https://github.com/paritytech/polkadot-sdk/blob/1e3b8e1639c1cf784eabf0a9afcab1f3987e0ca4/polkadot/node/network/bitfield-distribution/src/lib.rs#L135).
+If our implementation only supports version 2 and 3 messages, the subsystem should probably ignore peers that still use
+version 1 instead.
+
+The subsystem also needs to know the current and previous network grid topologies and the view of the node ("our view").
+
+[For each relay parent](https://github.com/paritytech/polkadot-sdk/blob/1e3b8e1639c1cf784eabf0a9afcab1f3987e0ca4/polkadot/node/network/bitfield-distribution/src/lib.rs#L161)
+the subsystem is instructed by the overseer to work on, it needs to maintain the following data:
+- the signing context (retrieved from runtime once)
+- the validator set (retrieved from runtime once)
+- any valid bitfield messages received from a validator (we can probably just store the network messages, since it's v2/v3 only)
+- messages sent to a peer for this relay parent (store the validator ID instead of the message since there can only be one per validator)
+- messages received from a peer for this relay parent (again, store validator ID, this is to avoid sending this peer a message for this validator)
+
+## Message Handling Logic
```

Comment:
LGTM, just this section, I don't think that is valuable to the design if we only have the links to the function defs if we don't give any explanation about them, so maybe we can remove this section.

Reply:
Yeah, that works as well!